code/TaxonomyTerm.php

```diff
@@ -125,6 +127,29 @@ public function onBeforeDelete()
             $term->delete();
         }
     }
+    /**
+     *  Set the URL segment allowing filtering by url slug
+     *
+     *  {@inheritDoc}
+     */
+    public function onBeforeWrite() {
+
+        parent::onBeforeWrite();
+
+        // Write the URLSegment to allow for Taxonomy navigation only if one is not supplied
+        if ($this->Name && $this->URLSegment == null)
+        {
+            $this->URLSegment = SiteTree::create()->generateURLSegment($this->Name);
+
+            //check if the URLSegment is unique otherwise add a prefix including the taxonomy type to distinguish
+            if (TaxonomyTerm::get()->filter(['URLSegment' => $this->URLSegment])->exclude(['ID' => $this->ID])->exists())
+            {
+                $this->URLSegment = SiteTree::create()->generateURLSegment($this->getTaxonomyType() . "-" .$this->Name);
```

Comment:
There's no guarantee this will be unique, so this should be in a while loop, ideally, and adding a random number or ID to the end.

Reply:
will do.